ml-proto/spec/check.ml

```diff
@@ -134,9 +134,9 @@ let rec check_expr c et e =
     check_expr_opt c (label c x) eo e.at
 
   | Br_if (x, eo, e) ->
+    check_expr_opt c et eo e.at;
```

Comment:
@sunfishcode Here might be one example for which the consumers expect different numbers of result values. 

```
(func (param $save i32)
  (block $top
    (i32.store (addr) (br_if $top (call $i32_side_effect_fn) (get_local $save)))))
```

I do wonder if we are overdoing the expressions optimizations. There might remain a lot of cases that still require writing state to locals between blocks and the stores/loads might still need to be optimized away well by the runtime compiler, so might the overall design be simpler without a value expression for br_if at all. There is a seemingly efficient wasm SSA pass, and from there it doesn't seem too hard to optimize inter-block copy operations away. I hope there is enough support to land this change, with more flexible checks, and perhaps this could all be re-visited later and compared to a minimalist design. Shall be interesting to see how useful this is in practice.